pandas/plotting/_matplotlib/core.py

```diff
@@ -1204,9 +1204,8 @@ def get_label(i):
 
         condition = (
             not self._use_dynamic_x()
-            and data.index.is_all_dates
-            and not self.subplots
-            or (self.subplots and self.sharex)
+            and (data.index.is_all_dates and self.use_index)
+            and (not self.subplots or (self.subplots and self.sharex))
```

Comment:
could you pls add descriptive comment for `condition`? seems quite complex now

Reply:
yeah this is getting very complex, pls comment , or use a helper variable to make this more clear / readable.

Reply:
Sure, done